### PR TITLE
HTCONDOR-1711: DAGMan direct produce creds branch

### PR DESCRIPTION
**Pull request by ColeBollig**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/version-history/feature-versions-23-x.rst [hunk 1]
```diff
@@ -21,7 +21,9 @@ New Features:
 
 Bugs Fixed:
 
-- None.
+- Fixed a bug where DAGMan would not produce job credentials when submitting jobs directly
```
Comment (johnkn) [inline docs/version-history/feature-versions-23-x.rst:24]: not a bug.

File: src/condor_dagman/dagman_main.cpp [hunk 2]
```diff
@@ -1364,28 +1367,21 @@ void main_pre_dc_init (int, char*[]) {
...
-	// Get the current directory
-	std::string currentDir = "";
-	if (condor_getcwd(currentDir)) {
-		currentDir += DIR_DELIM_STRING;
+	// Convert the DAGMan log file name to an absolute path if it's not one already
+	std::string fullLogFile;
+	const char* logFile = GetEnv("_CONDOR_DAGMAN_LOG");
+	if (logFile) {
+		if ( ! fullpath(logFile)) {
+			fullLogFile = dagman.workingDir + DIR_DELIM_STRING + logFile;
```
Comment (johnkn) [inline src/condor_dagman/dagman_main.cpp:1375]: should be dirscat

File: src/condor_dagman/dagman_main.cpp [hunk 3]
```diff
@@ -1364,28 +1367,21 @@ void main_pre_dc_init (int, char*[]) {
...
-	}
-
-	// Convert the DAGMan log file name to an absolute path if it's
-	// not one already
-	std::string newLogFile;
-	const char*	logFile = GetEnv("_CONDOR_DAGMAN_LOG");
-	if (logFile && !fullpath(logFile)) {
-		newLogFile = currentDir + logFile;
-		SetEnv("_CONDOR_DAGMAN_LOG", newLogFile.c_str());
+		fullLogFile = dagman.workingDir + DIR_DELIM_STRING + "default.dagman.out";
```
Comment (johnkn) [inline src/condor_dagman/dagman_main.cpp:1379]: dirscat

File: src/condor_utils/dagman_utils.cpp [hunk 4]
```diff
@@ -154,12 +154,15 @@ DagmanUtils::writeSubmitFile(DagmanOptions &options, str_list &dagFileAttrLines)
...
 	args.AppendArg("-p");
 	args.AppendArg("0");
-	args.AppendArg("-f");
-	args.AppendArg("-l");
-	args.AppendArg(".");
+	// -ld disables daemoncore creating debug log relative to $(LOG)
+	// Done so we don't override and can access $(LOG) for local daemon address files
+	args.AppendArg("-ld");
+	//======End DaemonCore Commands======
+
```
Comment (johnkn) [inline src/condor_utils/dagman_utils.cpp:165]: lets not change any of this yet.

**Review by johnkn — CHANGES_REQUESTED** [review]

File: src/condor_dagman/dagman_main.cpp (hunk 3 above)
Comment (ColeBollig) [reply src/condor_dagman/dagman_main.cpp]: `dirscat()` fails due to trailing path separator replaced with `dircat()` instead

**Review by johnkn — APPROVED** [review]
Looks good.

**Pull request merged**